All_Relationships/1.data-loader.py

```diff
@@ -0,0 +1,232 @@
+
+# In[ ]:
+
+get_ipython().magic(u"time filter_df = pd.read_table('/home/danich1/Documents/pubtator/data/pubtator-hetnet-tags.tsv.xz')")
+
+
+# In[ ]:
+
+get_ipython().magic(u"time grouped = filter_df.groupby('pubmed_id')")
+
+
+# In[ ]:
+
+# Please change to your local document here
+working_path = '/home/danich1/Documents/Database/pubmed_docs.xml'
+xml_parser = XMLMultiDocPreprocessor(
+    path= working_path,
+    doc='.//document',
+    text='.//passage/text/text()',
+    id='.//id/text()', tag_filter=set(filter_df['pubmed_id']))
+
+
+# In[ ]:
+
+dg_tagger = Tagger(grouped)
+
+
+# In[ ]:
+
+corpus_parser = CorpusParser(fn=dg_tagger.tag)
+document_chunk = []
+
+for document in tqdm.tqdm(xml_parser.generate()):
+    
+    document_chunk.append(document)
+
+    # chunk the data because snorkel cannot 
+    # scale properly yet
+    if len(document_chunk) >= 5e4:
+        corpus_parser.apply(document_chunk, parallelism=5, clear=False)
+        document_chunk = []
+    
+# If generator exhausts and there are still
+# document to parse
+if len(document_chunk) > 0:
+    corpus_parser.apply(data, parallelism=5, clear=False)
+    document_chunk = []
+
+
+# # Get each candidate relation
+
+# This block of code below is designed to gather and tag each sentence found. **Note**: This does include the title of each abstract.
+
+# In[ ]:
+
+chunk_size = 2e5
+
+
+# In[ ]:
+
+def insert_cand_to_db(extractor, sentences):
+    for split, sens in enumerate(sentences):
+        extractor.apply(sens, split=split, parallelism=5, clear=False)
+
+
+# In[ ]:
+
+def print_candidates(context_class, edge):
+    for i, label in enumerate(["Train", "Dev", "Test"]):
+        cand_len = session.query(context_class).filter(context_class.split == i).count()
+        print("Number of Candidates for {} edge and {} set: {}".format(edge, label, cand_len))
+
+
+# In[ ]:
+
+#This specifies the type of candidates to extract
+DiseaseGene = candidate_subclass('DiseaseGene', ['Disease', 'Gene'])
+dge = PretaggedCandidateExtractor(DiseaseGene, ['Disease', 'Gene'])
+
+GeneGene = candidate_subclass('GeneGene', ['Gene1', 'Gene2'])
+gge = PretaggedCandidateExtractor(GeneGene, ['Gene', 'Gene'])
+
+CompoundGene = candidate_subclass('CompoundGene', ['Compound', 'Gene'])
+cge = PretaggedCandidateExtractor(CompoundGene, ['Compound', 'Gene'])
+
+CompoundDisease = candidate_subclass('CompoundDisease', ['Compound','Disease'])
+cde = PretaggedCandidateExtractor(CompoundDisease, ['Compound', 'Disease'])
+
+
+# In[ ]:
+
+# set the seed for reproduction
+np.random.seed(100)
+total_sentences = 61615305
+
+
+# In[ ]:
+
+category_list = np.random.choice([0,1,2], total_sentences, p=[0.7,0.2,0.1])
+
+
+# In[ ]:
+
+# Divide the sentences into train, dev and test sets
+   
+#Grab the sentences!!!
+train_sens = set()
+dev_sens = set()
+test_sens = set()
+
+offset = 0
+category_index = 0
+has_docs = True
+#divde and insert into the database
+while has_docs:
```

Comment:
What is changing in each iteration of `while has_docs`? I think we may be able to make this clearer.